src/Core/Checkout/Cart/Event/CartChangedEvent.php

```diff
@@ -3,38 +3,33 @@
 namespace Shopware\Core\Checkout\Cart\Event;
 
 use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\Event\ShopwareSalesChannelEvent;
 use Shopware\Core\Framework\Log\Package;
 use Shopware\Core\System\SalesChannel\SalesChannelContext;
 use Symfony\Contracts\EventDispatcher\Event;
 
 #[Package('checkout')]
-class CartChangedEvent extends Event
+class CartChangedEvent extends Event implements ShopwareSalesChannelEvent, CartEvent
 {
-    /**
-     * @var Cart
-     */
-    protected $cart;
-
-    /**
-     * @var SalesChannelContext
-     */
-    protected $context;
-
     public function __construct(
-        Cart $cart,
-        SalesChannelContext $context
+        protected readonly Cart $cart,
+        protected readonly SalesChannelContext $salesChannelContext,
     ) {
-        $this->cart = $cart;
-        $this->context = $context;
     }
 
     public function getCart(): Cart
     {
         return $this->cart;
     }
 
-    public function getContext(): SalesChannelContext
+    public function getContext(): Context
+    {
+        return $this->salesChannelContext->getContext();
+    }
```

Comment:
Changing the return of `getContext` is also a breaking change. I would suggest this:

```suggestion
    /**
     * @deprecated tag:v6.7.0 - Use getSalesChannelContext() instead
     */
    public function getContext(): SalesChannelContext
    {
        return $this->getSalesChannelContext();
    }
```

Then once it's removed, `getContext()` can be updated to return the `Context`

Reply:
Changed, however this means CartChangedEvent cannot implement ShopwareSalesChannelEvent yet.